core/src/main/scala/org/apache/spark/storage/DiskBlockObjectWriter.scala

```diff
@@ -288,3 +288,19 @@ private[spark] class DiskBlockObjectWriter(
     bs.flush()
   }
 }
+
+private[spark] object DiskBlockObjectHelper extends Logging {
+
+  /**
+   * Reverts writes that haven't been committed yet and delete the object file hold by the
+   * DiskBlockObjectWriter.
+   */
+  def deleteAbnormalDiskBlockObjectFile(writer: DiskBlockObjectWriter): Unit = {
```

Comment:
Why not keep it as a member function of `DiskBlockObjectWriter` ?

Reply:
@Ngone51 That's a good question, but I think it's strange for an `OutputStream( DiskBlockObjectWriter extends OutputStream)` to delete itself. 

For example, `o.a.h.fs.FileSystem` can `create` a `FSDataOutputStream`,  although the `FSDataOutputStream` has the capabilities of `write data` and `close itself`, but we still need to use the `o.a.h.fs.FileSystem` to delete the file.

Of course, there are other ways of API design, such as `java.io.File` can `createNewFile` and the `java.io.File` instance can `delete` itself.

At present, I choose the former because `DiskBlockObjectWriter` is a `OutputStream` rather than a `File` ,  but I also accept it as a member function of `DiskBlockObjectWriter`.

Do you think `keep it as a member function of DiskBlockObjectWriter` is better?

Reply:
Ok, I got your point. Make sense to me.

But, to be honest, I think the usage of the helper method `deleteAbnormalDiskBlockObjectFile` is less straightforward compares to the current one.

Reply:
Got it ~  I will close this pr